### DISCUSSION TITLE
[Feature Request] How to Transfer Entire Book Collection from one PC To Another Regardless of OS??

### DISCUSSION THREAD

**Discussion by mbennette68** [edrlab/thorium-reader#2680] [Ideas]
Hello,

I see no feature in Thorium Reader that allows for the transfer of the **_entire_** book collection from one machine to another.
An export feature would be great, or at least easy to follow instructions on how to manually do it, for any OS would be great.

Thank you,
MB

**Comment by llemeurfr** [comment]
Sure, it would be useful in case of a migration from one computer to another. The publications are stored in a well-known directory, we'll try to document it in https://thorium.edrlab.org/en/docs/. 

Note: Having a UI for exporting the whole catalog is not a priority currently.

**Comment by gautierchomel** [comment]
From internal discussions: 

> The user can very well copy the electron useData directory which depends on the OS and copy it to the same location on the target machine, then start Thorium. This should work. Now we do not guarantee this method, because we do not know the limits of this practice, nor the potential problems of migrating the thorium db (which would be too old, but copied in Thorium3).
> Likewise, synchronizing the userData folder using rsync/dropbox works but be careful about reconciling modified files between the two machines. It's a complex problem. 

Therefore, a [support faq entry](https://thorium.edrlab.org/en/th3/400_ressources/430_faq/#localStorage) was added with stable manual transfer workaround.

**Reply by inklesspen** [reply]
> Therefore, a [support faq entry](https://thorium.edrlab.org/en/th3/400_ressources/430_faq/#localStorage) was added with stable manual transfer workaround.

This link no longer works.

**Comment by danielweck** [comment]
What about migrating OPDS feeds? (possibly with credentials / access tokens?)
What about stored / cached LCP passphrases?
What about annotations / bookmarks that are attached to stored publications?
What about reading progress?

I am moving this issue to a discussion, because as it stands now, this is not an actionable issue in the tracker, we must dig a little deeper in order to flesh out problem details, and work out a solution.

**Comment by mininanospork** [comment]
Thank you for documenting where books may be found in the online documentation. 
Yes. Good work to render book representations inside the Thorium Reader App.

How ever not good at all follows.

Thorium provides anti-features by removing the name of the book from the OS file name and creating support files that are not incorporated in the native OS file explorer for viewing and management of the book file itself. This forces the user to be dependent on the Thorium Reader to manage their own books, actively preventing all other methods of management. Worse the user must wait on the Thorium Development team to get around to thinking about doing something in this area. 

epub files have distinct names. Would be far better for Thorium Reader to pick a file name scheme to consistently follow such as (Title - Author - Date - publisher - ISBN) or even better let the user define in configuration and update all files for the user based on the metadata you already store to enable their own visibility, since there is currently no standard for epub file names. Further the meta data scheme Thorium Reader uses should be documented so that folks who want to incorporate Thorium reader into one or more personalized work flows of use would easily be able to.

For example it may be desirable for users to have annotations flow into and out of Thorium readers into other systems or services they may use such as Obsidian (local) or Notion (online) or to publish or create mini-reviews to various social media services like Good Reads and Others.
